bindings/c/Cargo.toml

```diff
@@ -36,23 +36,43 @@ cbindgen = "0.26.0"
 [dependencies]
 bytes = "1.4.0"
 # this crate won't be published, we always use the local version
-opendal = { version = ">=0", path = "../../core", features = [
-  "layers-blocking",
-  # These are default features before v0.46. TODO: change to optional features #4313
-  "services-azblob",
-  "services-azdls",
-  "services-cos",
-  "services-fs",
-  "services-gcs",
-  "services-ghac",
-  "services-http",
-  "services-ipmfs",
-  "services-memory",
-  "services-obs",
-  "services-oss",
-  "services-s3",
-  "services-webdav",
-  "services-webhdfs",
-  "services-azfile",
-] }
+opendal = { version = ">=0", path = "../../core", default-features = false, features = ["layers-blocking"] }
 tokio = { version = "1.27", features = ["fs", "macros", "rt-multi-thread"] }
+
+[features]
+default = []
+# Define features that enable corresponding opendal service features
+services-azblob = ["opendal/services-azblob"]
+services-azdls = ["opendal/services-azdls"]
+services-cos = ["opendal/services-cos"]
+services-fs = ["opendal/services-fs"]
+services-gcs = ["opendal/services-gcs"]
+services-ghac = ["opendal/services-ghac"]
+services-http = ["opendal/services-http"]
+services-ipmfs = ["opendal/services-ipmfs"]
+services-memory = ["opendal/services-memory"]
+services-obs = ["opendal/services-obs"]
+services-oss = ["opendal/services-oss"]
+services-s3 = ["opendal/services-s3"]
+services-webdav = ["opendal/services-webdav"]
+services-webhdfs = ["opendal/services-webhdfs"]
+services-azfile = ["opendal/services-azfile"]
+
+# Feature to enable all services (optional)
+services-all = [
```

Comment:
I wish we can just remove this.

Reply:
but other bindings, like zig, will meet problems.

Reply:
Maybe we can remove the dependencies of this feature first.

Reply:
If we have nice features support, maybe we can change the line here:

https://github.com/apache/opendal/blob/da026ef80787ce85af0c6deb2efdbe919838067a/bindings/zig/build.zig#L65

Reply:
oh, you mean just enable the needed zig feature?